cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -48,6 +48,107 @@ T __device__ inline normalize_nans_and_zeros(T const& key)
   return key;
 }
 
+__device__ inline uint32_t rotate_bits_left(uint32_t x, uint32_t r)
```

Comment:
I moved some common functions/classes into this header so that it can be re-used by the SHA hash in a follow-up PR. This code was not edited from the previous file except to replace some `int8` arguments with `uint32_t` arguments for the bit rotation functions. This change aligns with the input data types and expected types for the CUDA intrinsics `__funnelshift_l` / `__funnelshift_r`.